### ISSUE TITLE
NestedError: topic/partition change check failed error

### ISSUE THREAD

**Issue by YonatanKra** [SOHU-Co/kafka-node#1228]
## Questions?

This suddenly happened.  I can't replicate this, as it happened only once and my code is pretty straight forward - I raise a a few consumerGroups and just consume data.

Can you please shed some light as to what might cause this? 
Since it's not replicated easily, I'd like a direction where to look at (is it my JS? the Kafka Broker itself)?

* Node version: 10.15.3
* Kafka-node version: 4.0.1
* Kafka version: 2.2.0

### For specific cases also provide
* Number of Brokers: 1
* Number partitions for topic: 1

**Comment by hyperlink** [comment]
What was the error? can you share a full stack trace?

**Comment by syberstudios** [comment]
> What was the error? can you share a full stack trace?

Having the same problem here.

/home/node/app/node_modules/kafka-node/lib/consumerGroup.js:295
      return this.emit('error', new NestedError('topic/partition change check failed', error));
                                ^
NestedError: topic/partition change check failed
at _checkTopicPartitionChange (/home/node/app/node_modules/kafka-node/lib/consumerGroup.js:295:37)
at client.loadMetadataForTopics (/home/node/app/node_modules/kafka-node/lib/consumerGroup.js:323:14)
at async.series (/home/node/app/node_modules/kafka-node/lib/kafkaClient.js:892:7)
at /home/node/app/node_modules/async/dist/async.js:3888:9
at /home/node/app/node_modules/async/dist/async.js:473:16
at iterateeCallback (/home/node/app/node_modules/async/dist/async.js:988:17)
at /home/node/app/node_modules/async/dist/async.js:969:16
at /home/node/app/node_modules/async/dist/async.js:3885:13
at Timeout.setTimeout (/home/node/app/node_modules/kafka-node/lib/kafkaClient.js:487:5)
at ontimeout (timers.js:498:11)
Caused By: TimeoutError: Request timed out after 30000ms
at new TimeoutError (/home/node/app/node_modules/kafka-node/lib/errors/TimeoutError.js:6:9)
at Timeout.setTimeout (/home/node/app/node_modules/kafka-node/lib/kafkaClient.js:487:14)
at ontimeout (timers.js:498:11)
at tryOnTimeout (timers.js:323:5)
at Timer.listOnTimeout (timers.js:290:5)

**Comment by hyperlink** [comment]
This is a scheduled function that checks if there are new partitions in the topics being consumed. Looks like it failed to get the new metadata in your case. Perhaps a network issue to the broker?

**Comment by syberstudios** [comment]
Probably because I left it open and closed the lid of the laptop over night. Thank you, it makes sense now. :smile:

**Comment by thynson** [comment]
I encouter this bug too.

**Comment by alex3165** [comment]
I am having the same issue with the following stack trace:
```
NestedError: topic/partition change check failed
    at /app/node_modules/kafka-node/lib/consumerGroup.js:295:37
    at /app/node_modules/kafka-node/lib/consumerGroup.js:323:14
    at KafkaClient.loadMetadataForTopics (/app/node_modules/kafka-node/lib/kafkaClient.js:891:12)
    at ConsumerGroup._checkTopicPartitionChange (/app/node_modules/kafka-node/lib/consumerGroup.js:321:15)
    at Timeout._onTimeout (/app/node_modules/kafka-node/lib/consumerGroup.js:292:12)
    at listOnTimeout (internal/timers.js:531:17)
    at processTimers (internal/timers.js:475:7)
Caused By: BrokerNotAvailableError: Broker not available (loadMetadataForTopics)
    at new BrokerNotAvailableError (/app/node_modules/kafka-node/lib/errors/BrokerNotAvailableError.js:11:9)
    at KafkaClient.loadMetadataForTopics (/app/node_modules/kafka-node/lib/kafkaClient.js:891:21)
    at ConsumerGroup._checkTopicPartitionChange (/app/node_modules/kafka-node/lib/consumerGroup.js:321:15)
    at Timeout._onTimeout (/app/node_modules/kafka-node/lib/consumerGroup.js:292:12)
    at listOnTimeout (internal/timers.js:531:17)
    at processTimers (internal/timers.js:475:7)
```

I suspect my Broker was flaky for some time and the request failed, maybe it would be nice to have a retry logic on that case

**Comment by thynson** [comment]
@alex3165 Check PR #1281 if it helps you.
